include/unifex/on.hpp

```diff
@@ -42,10 +42,11 @@ namespace _on {
         (requires sender<Sender> AND scheduler<Scheduler> AND //
           (!tag_invocable<_fn, Sender, Scheduler>))
     auto operator()(Sender&& sender, Scheduler&& scheduler) const {
-      return with_query_value(
-          sequence(schedule(), (Sender&&)sender),
+      return sequence(schedule(scheduler),
```

Comment:
```suggestion
      auto sched = schedule(scheduler);
      return sequence(std::move(sched),
```

You're potentially moving from `scheduler` in the same expression. That's potential UB considering the indeterminate order of function argument evaluation.

Reply:
yes, I looked at that and did not remember indeterminate eval. :(

used your suggestion but with s/sched/sndr/